app/parsers/katz.py

```diff
@@ -0,0 +1,124 @@
+                            mk_section.logon_time = logon_time.group(1)
+                        sid = re.match(r'^\s*SID\s*:\s*([^\r\n]*)', line)
+                        if sid:
+                            mk_section.sid = sid.group(1)
+                        continue
+
+                if line.startswith('['):
+                    # this might indicate the start of a new account
+                    if 'Username' in package and package['Username'] != '(null)' and \
+                            (('Password' in package and package['Password'] != '(null)') or 'NTLM' in package):
+                        mk_section.packages[package_name].append(package)
+
+                    # reset the package
+                    package = {}
+                    pass
+                elif line.startswith('*'):
+                    m = re.match(r'\s*\* (.*?)\s*: (.*)', line)
+                    if m:
+                        package[m.group(1)] = m.group(2)
+
+                elif line:
+                    # parse out the new section name
+                    match_group = re.match(r'([a-z]+) :', line)
+                    if match_group:
+                        # this is the start of a new ssp
+                        # save the current ssp if necessary
+                        if 'Username' in package and package['Username'] != '(null)' and \
+                                (('Password' in package and package['Password'] != '(null)') or 'NTLM' in package):
+                            mk_section.packages[package_name].append(package)
+
+                        # reset the package
+                        package = {}
+
+                        # get the new name
+                        package_name = match_group.group(1)
+
+            # save the current ssp if necessary
+            if 'Username' in package and package['Username'] != '(null)' and \
+                    (('Password' in package and package['Password'] != '(null)') or 'NTLM' in package):
+                mk_section.packages[package_name].append(package)
+
+            # save this section
+            if mk_section.packages:
+                creds.append(mk_section)
+
+        return creds
+
+    def parse(self, blob):
+        relationships = []
+        try:
+            parse_data = self.parse_kats(blob)
+            for match in parse_data:
+                if self.parse_mode in match.packages:
+                    for mp in self.mappers:
+                        relationships.append(
+                            Relationship(source=(mp.get('source'), match.packages[self.parse_mode][0]['Username']),
+                                         edge=mp.get('edge'),
+                                         target=(mp.get('target'), match.packages[self.parse_mode][0]['Password']))
+                        )
+        except Exception as error:
+            self.log.warning('Mimikatz parser encountered an error - {}. Continuing...'.format(error))
+        return relationships
```

Comment:
Well done, easily readable and to the point

Reply:
The error handling is a bit generic, but it should cover all potential issues and log the problem.